pkg/restapi/vault/operation/operations.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package operation
+
+import (
+	"net/http"
+
+	"github.com/trustbloc/edge-service/pkg/internal/common/support"
+)
+
+const (
+	operationID             = "/vaults"
+	createVaultPath         = operationID
+	saveDocPath             = operationID + "/{vaultID}/docs"
+	getDocPath              = operationID + "/{vaultID}/docs/{docID}"
+	createAuthorizationPath = operationID + "/{vaultID}/authorizations"
+	getAuthorizationPath    = operationID + "/{vaultID}/authorizations/{authID}"
+	deleteAuthorizationPath = operationID + "/{vaultID}/authorizations/{authID}"
+)
+
+// Operation defines handlers for vault service.
+type Operation struct{}
+
+// Config defines configuration for vault operations.
+type Config struct{}
+
+// New returns operation instance.
+func New(cfg *Config) (*Operation, error) {
+	return &Operation{}, nil
+}
+
+// GetRESTHandlers get all controller API handler available for this service.
+func (o *Operation) GetRESTHandlers() []support.Handler {
+	return []support.Handler{
+		support.NewHTTPHandler(createVaultPath, http.MethodPost, o.CreateVault),
+		support.NewHTTPHandler(saveDocPath, http.MethodPost, o.SaveDoc),
+		support.NewHTTPHandler(getDocPath, http.MethodGet, o.GetDoc),
```

Comment:
This changed to `GET /vaults/{vaultID}/docs/{docID}/metadata` in the latest revision. The reason is we don't have an immediate use case for a user to read back their own documents decrypted.

The document metadata endpoint is a sort of utility method that allows internal orchestration between the components with the aim of hiding some details from the user (eg. EDV doc URIs).

Reply:
Fixed #573